File: branches/Medusa/Source/Fantasy.Jobs/JobRuntimeStateReader.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ComponentModel;
using Fantasy.XSerialization;

namespace Fantasy.Jobs
{
    public class JobRuntimeStateReader :ObjectWithSite, ITagValueProvider 
    {
        #region ITagValueProvider Members

        public char Prefix
        {
            get { return '#'; }
        }

        public string GetTagValue(string tag, IDictionary<string, object> context)
        {
            string rs;
            this.TryGetValue(tag, context, out rs);
            return rs;
        }


        private bool TryGetValue(string name, IDictionary<string, object> context, out string value)
        {
            bool rs = false;
            value = null;
            if (this.Site != null)
            {

                string[] names = name.Split(new char[] { '.' }, 2);
                name = names[0];

                string meta = names.Length > 1 ? names[1] : null;

                IJob job = (Job)this.Site.GetService(typeof(IJob));

                object o;

                rs = job.RuntimeStatus.TryGetValue(name, out o);

                if (rs)
                {
                    if (o is TaskItem)
                    {
                        TaskItem item = (TaskItem)o;
                        value = meta != null ? item[meta] : item.Name;
                    }
                    else if(meta != null)
                    {
                        TaskItem item = job.GetEvaluatedItemByName((string)o);
                        if(item != null)
                        {
                            value = item[meta];
                        }
                    }
                    else if(o != null)
                    {
                        TypeConverter cvt = XHelper.Default.CreateXConverter(o.GetType());
                        value = cvt.ConvertToString(o); 
                    }
                }
                
            }

           
            return rs;
        }

        public bool HasTag(string tag, IDictionary<string, object> context)
        {

            string value;
            return this.TryGetValue(tag, context, out value);
           
        }

        #endregion

        #region ITagValueProvider Members


        public bool IsEnabled(IDictionary<string, object> context)
        {
            return (bool)context.GetValueOrDefault("EnableTaskItemReader", true) &&
                this.Site != null && this.Site.GetService(typeof(IJob)) != null;
        }

        #endregion
    }
}

File: AutoConverted/Fantasy.Jobs/JobRuntimeStateReader.java
package Fantasy.Jobs;

import Fantasy.XSerialization.*;

public class JobRuntimeStateReader extends ObjectWithSite implements ITagValueProvider
{
//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#region ITagValueProvider Members

	public final char getPrefix()
	{
		return '#';
	}

	public final String GetTagValue(String tag, java.util.Map<String, Object> context)
	{
		String rs = null;
		RefObject<String> tempRef_rs = new RefObject<String>(rs);
		this.TryGetValue(tag, context, tempRef_rs);
		rs = tempRef_rs.argvalue;
		return rs;
	}


	private boolean TryGetValue(String name, java.util.Map<String, Object> context, RefObject<String> value)
	{
		boolean rs = false;
		value.argvalue = null;
		if (this.Site != null)
		{

			String[] names = name.split(new char[] { '.' }, 2);
			name = names[0];

			String meta = names.length > 1 ? names[1] : null;

			IJob job = (Job)this.getSite().GetService(IJob.class);

			Object o = null;

			RefObject<Object> tempRef_o = new RefObject<Object>(o);
			rs = job.getRuntimeStatus().TryGetValue(name, tempRef_o);
			o = tempRef_o.argvalue;

			if (rs)
			{
				if (o instanceof TaskItem)
				{
					TaskItem item = (TaskItem)o;
					value.argvalue = meta != null ? item.getItem(meta) : item.getName();
				}
				else if(meta != null)
				{
					TaskItem item = job.GetEvaluatedItemByName((String)o);
					if(item != null)
					{
						value.argvalue = item.getItem(meta);
					}
				}
				else if(o != null)
				{
					TypeConverter cvt = XHelper.Default.CreateXConverter(o.getClass());
					value.argvalue = cvt.ConvertToString(o);
				}
			}

		}


		return rs;
	}

	public final boolean HasTag(String tag, java.util.Map<String, Object> context)
	{

		String value = null;
		RefObject<String> tempRef_value = new RefObject<String>(value);
		boolean tempVar = this.TryGetValue(tag, context, tempRef_value);
		value = tempRef_value.argvalue;
		return tempVar;

	}

//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#endregion

//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#region ITagValueProvider Members


	public final boolean IsEnabled(java.util.Map<String, Object> context)
	{
		return (boolean)context.GetValueOrDefault("EnableTaskItemReader", true) && this.Site != null && this.Site.GetService(IJob.class) != null;
	}

//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#endregion
}